### DISCUSSION TITLE
2009/07/30/labview-scripting-%E5%BC%80%E5%A7%8B%E5%AF%B9%E5%AE%A2%E6%88%B7%E5%BC%80%E6%94%BE

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#270] [Announcements]
# 2009/07/30/labview-scripting-%E5%BC%80%E5%A7%8B%E5%AF%B9%E5%AE%A2%E6%88%B7%E5%BC%80%E6%94%BE

使用LabVIEW Scripting (也叫VI Scripting) 可以通过编程来创建和修改VI代码。这不算是一个新功能，LabVIEW 6.0 就加入了一些简单的 LabVIEW Scripting 功能。但是长期以来，LabVIEW普通用户只能使用一些基础的 LabVIEW Scripting 功能。那些强大的API只是针对NI内部用户或联盟商的，普通用户很难拿到授权。也许这是因为LabVIEW Scripting的功能太过强大了，公司总是担心把它公布出来会有意想不到的风险。

https://blog.qizhen.xyz/2009/07/30/labview-scripting-%E5%BC%80%E5%A7%8B%E5%AF%B9%E5%AE%A2%E6%88%B7%E5%BC%80%E6%94%BE

**Comment by ruanqizhen** [comment]
之前博客下的留言：

彬
2009年09月13日 22:19 编辑
阮大侠： 谢谢你的回答，但由此也就决定了LabVIEW Scripting用在实际工程上的可能性大大削弱了，因为它的高级功能不能使用，那就跟普通的属性节点和方法节点没有区别了。

回复
Qizhen
2009年09月10日 14:21 编辑
LabVIEW Scripting 的高级功能只被加到了LabVIEW的编辑环境中，并没有被合成到运行环境中。Add Button.vi做成EXE，就脱离了编辑环境，就不能运行了。VI有些属性是可以在运行时改变的，比如caption、颜色等，但有些属性不能在运行时改变，比如label，控件类型等。所以使用LabVIEW Scripting要考虑使用的场合。比如目标VI如果是在运行时，就要注意不能改变他的某些属性。如果应用程序需要脱离LabVIEW编辑环境运行，就要避免使用一些高级功能。

回复
彬
2009年09月10日 14:08 编辑
阮大侠： 我试过用VI Scripting工具包编写了一些极其简单的程序，比如：我写了一个程序叫Add Button.vi，它能通过输入的VI路径（此路径对应的VI暂且称为A.vi），为A.vi动态添加一个Button。但发现有几个问题： 1.如果在A.vi在Edit Mode状态下，而且Add Button.vi没有build成exe时，可以实现预想功能。 2.如果在A.vi在Run Mode状态下，无论Add Button.vi是否build成exe，都不能实现预想功能。 3.如果在A.vi在Edit Mode状态下，而Add Button.vi已经build成exe，还是不能实现预想功能，并且程序进入Add Button.vi中相关的VI Sctriping工具包的vi，程序会卡死在那里，而且没有任何错误代码返回。 由以上实验总结出：VI Scripting只适合给LabVIEW编写辅助编程工具，如NI的Project Explorer等等（还是处于vi状态，并非exe），并不适合工程中编写最终产品。 我这样的理解不知道对不对，请阮大侠指教一下。

回复
l
2009年07月30日 13:35 编辑
这个很不错，可以通过LabVIEW脚本来控制LabVIEW生成代码，感觉和Office里面的VBA差不多了。

回复
永青
2009年07月30日 13:26 编辑
我来做个广告，以前写过的一片关于VI Scripting的文章：http://yeyongq.spaces.live.com/blog/cns!A52629657552983C!914.entry

回复

**Comment by ruanqizhen** [comment]
之前博客下的留言：

BO
2009年08月5日 00:10 编辑
这幅图可以用来给vi script做广告了。

回复
Qizhen
2009年08月4日 15:52 编辑
没什么实用，只是看着爽。

回复
Jay
2009年08月4日 15:10 编辑
连线弄得这么绕，有什么特殊用处吗？弯折的直线不是挺好的嘛

回复
